numba/unicode.py

```diff
@@ -1267,6 +1267,56 @@ def impl(a):
         return impl
 
 
+# https://github.com/python/cpython/blob/1d4b6ba19466aba0eb91c4ba01ba509acf18c723/Objects/unicodeobject.c#L10765-L10774    # noqa: E501
+@overload_method(types.UnicodeType, 'capitalize')
+def unicode_capitalize(data):
+    """Implements str.capitalize()"""
+    def impl(data):
+        length = len(data)
+        if length == 0:
+            return _empty_string(data._kind, length, data._is_ascii)
+
+        if data._is_ascii:
+            res = _empty_string(data._kind, length, 1)
+            code_point = _get_code_point(data, 0)
+            _set_code_point(res, 0, _Py_TOUPPER(code_point))
```

Comment:
In terms of efficiency, inside this section, it's probably faster to use the `_Py_IS<FOO>` methods to check if it's already `FOO` and then only translate the code point as needed? (like in the bytes method ref above).